tensorflow/core/kernels/cwise_ops_test.cc

```diff
@@ -29,7 +29,16 @@ static Graph* Unary(const string& func, int num, DataType dtype) {
   Graph* g = new Graph(OpRegistry::Global());
   Tensor data(dtype, TensorShape({64, 64, num / (64 * 64)}));
   CHECK_GT(data.NumElements(), 0);
-  data.flat<float>().setRandom();
+  switch (dtype) {
```

Comment:
ok yes since there were only two types we were checking... but yes i prefer templates too. Sending a patch soon.